File: Speakers.Domain/SpeakingEngagement.cs
using NodaTime;
using System;
using System.Collections.Generic;
using System.Text;

namespace Speakers.Domain
{
    public class SpeakingEngagement
    {
        public Guid Id { get; set; }
        public string SpeakerName { get; set; } = "";
        public string TalkName { get; set; } = "";
        public string HostGroupName { get; set; } = "";
        public LocalDateTime DateTimeStart { get; set; }
        public LocalDateTime DateTimeEnd { get; set; }
        public string City { get; set; } = "";
        public string State { get; set; } = "";
        public int Attendance { get; set; }
    }
}

File: Speakers.UI/src/components/SpeakingEngagementsList/SpeakingEngagement.js
import React from 'react';
import PropTypes from 'prop-types';


const SpeakingEngagement = ({
  speakerName, talkName, hostGroupName, startDate, endDate, city, state, attendance, id,
}) => (
    <li>{startDate}. {speakerName} | {talkName} | {hostGroupName}
      | {endDate} | {city} | {state} | {attendance} | {id}</li>);

SpeakingEngagement.propTypes = {
  speakerName: PropTypes.string.isRequired,
  attendance: PropTypes.number,
  talkName: PropTypes.string.isRequired,
  startDate: PropTypes.string.isRequired,
  endDate: PropTypes.string.isRequired,
  hostGroupName: PropTypes.string,
  city: PropTypes.string,
  state: PropTypes.string,
  id: PropTypes.string,
};

export default SpeakingEngagement;
